File: rust/src/models/volume_plan_pricing.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::tier_row::TierRow;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct VolumePlanPricing {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub block_size: Option<i32>,

    pub tiers: Vec<TierRow>,
}

impl VolumePlanPricing {
    pub fn new(tiers: Vec<TierRow>) -> Self {
        Self {
            block_size: None,
            tiers,
        }
    }
}

File: typescript/src/models/volumePlanPricing.ts
// this file is @generated
import { type TierRow, TierRowSerializer } from "./tierRow";

export interface VolumePlanPricing {
  blockSize?: number | null;

  tiers: TierRow[];
}

export const VolumePlanPricingSerializer = {
  _fromJsonObject(object: any): VolumePlanPricing {
    return {
      blockSize: object["block_size"],
      tiers: object["tiers"].map((item: any) => TierRowSerializer._fromJsonObject(item)),
    };
  },

  _toJsonObject(self: VolumePlanPricing): any {
    return {
      block_size: self.blockSize,
      tiers: self.tiers.map((item: any) => TierRowSerializer._toJsonObject(item)),
    };
  },
};
